verticapy/utilities.py

```diff
@@ -449,168 +558,239 @@ def drop(name: str = "", method: str = "auto", raise_error: bool = False, **kwds
 
 
 # ---#
-def readSQL(query: str, time_on: bool = False, limit: int = 100):
-    """
-	---------------------------------------------------------------------------
-	Returns the result of a SQL query as a tablesample object.
-
-	Parameters
-	----------
-	query: str, optional
-		SQL Query.
-	time_on: bool, optional
-		If set to True, displays the query elapsed time.
-	limit: int, optional
-		Number maximum of elements to display.
-
- 	Returns
- 	-------
- 	tablesample
- 		Result of the query.
-	"""
-    # Saving information to the query profile table
-    save_to_query_profile(
-        name="readSQL",
-        path="utilities",
-        json_dict={"query": query, "time_on": time_on, "limit": limit,},
-    )
-    # -#
-    check_types(
-        [
-            ("query", query, [str]),
-            ("time_on", time_on, [bool]),
-            ("limit", limit, [int, float]),
-        ]
-    )
-    while len(query) > 0 and query[-1] in (";", " "):
-        query = query[:-1]
-    count = executeSQL(
-        "SELECT /*+LABEL('utilities.readSQL')*/ COUNT(*) FROM ({}) VERTICAPY_SUBTABLE".format(
-            query
-        ),
-        method="fetchfirstelem",
-        print_time_sql=False,
-    )
-    sql_on_init = verticapy.options["sql_on"]
-    time_on_init = verticapy.options["time_on"]
-    try:
-        verticapy.options["time_on"] = time_on
-        verticapy.options["sql_on"] = False
-        try:
-            result = to_tablesample("{} LIMIT {}".format(query, limit))
-        except:
-            result = to_tablesample(query)
-    except:
-        verticapy.options["time_on"] = time_on_init
-        verticapy.options["sql_on"] = sql_on_init
-        raise
-    verticapy.options["time_on"] = time_on_init
-    verticapy.options["sql_on"] = sql_on_init
-    result.count = count
-    if verticapy.options["percent_bar"]:
-        vdf = vDataFrameSQL("({}) VERTICAPY_SUBTABLE".format(query))
-        percent = vdf.agg(["percent"]).transpose().values
-        for column in result.values:
-            result.dtype[column] = vdf[column].ctype()
-            result.percent[column] = percent[vdf.format_colnames(column)][0]
-    return result
-
-
-# ---#
-def get_data_types(expr: str, column_name: str = ""):
+def get_data_types(
+    expr: str = "",
+    column: str = "",
+    table_name: str = "",
+    schema: str = "public",
+    usecols: list = [],
+):
     """
 ---------------------------------------------------------------------------
 Returns customized relation columns and the respective data types.
 This process creates a temporary table.
 
+If table_name is defined, the expression is ignored and the function
+returns the table/view column names and data types.
+
 Parameters
 ----------
-expr: str
-	An expression in pure SQL.
-column_name: str, optional
-	If not empty, it will return only the data type of the input column if it
-	is in the relation.
+expr: str, optional
```

Comment:
Not optional I think.

Reply:
Based on the below logic, specifying either of  expr or table_name would be enough.